internal/cli/api.go

```diff
@@ -0,0 +1,257 @@
+
+	return cmd
+}
+
+func apiCmdRun(cli *cli, inputs *apiCmdInputs) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := inputs.fromArgs(args, cli.tenant); err != nil {
+			return fmt.Errorf("failed to parse command inputs: %w", err)
+		}
+
+		if inputs.Method == http.MethodDelete && !cli.force && canPrompt(cmd) {
+			message := "Are you sure you want to proceed? Deleting is a destructive action."
+			if confirmed := prompt.Confirm(message); !confirmed {
+				return nil
+			}
+		}
+
+		var response *http.Response
+		if err := ansi.Waiting(func() error {
+			request, err := http.NewRequestWithContext(
+				cmd.Context(),
+				inputs.Method,
+				inputs.URL.String(),
+				inputs.Data,
+			)
+			if err != nil {
+				return err
+			}
+
+			bearerToken := cli.config.Tenants[cli.tenant].AccessToken
+			request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", bearerToken))
+			request.Header.Set("Content-Type", "application/json")
+			request.Header.Set("User-Agent", fmt.Sprintf("%s/%s", userAgent, strings.TrimPrefix(buildinfo.Version, "v")))
+
+			if cli.debug {
+				cli.renderer.Infof("[%s]: %s", request.Method, request.URL.String())
+			}
+
+			response, err = http.DefaultClient.Do(request)
+			return err
+		}); err != nil {
+			return fmt.Errorf("failed to send request: %w", err)
+		}
+		defer response.Body.Close()
+
+		rawBodyJSON, err := io.ReadAll(response.Body)
+		if err != nil {
+			return err
+		}
+
+		if len(rawBodyJSON) == 0 {
+			if cli.debug {
+				cli.renderer.Infof("Response body is empty.")
+			}
+			return nil
+		}
+
+		var prettyJSON bytes.Buffer
+		if err := json.Indent(&prettyJSON, rawBodyJSON, "", "  "); err != nil {
+			return fmt.Errorf("failed to prepare json output: %w", err)
+		}
+
+		cli.renderer.Output(ansi.ColorizeJSON(prettyJSON.String(), false))
+
+		return nil
+	}
+}
+
+func (i *apiCmdInputs) fromArgs(args []string, domain string) error {
+	i.parseRaw(args)
+
+	if err := i.validateAndSetMethod(); err != nil {
+		return err
+	}
+
+	if err := i.validateAndSetData(); err != nil {
+		return err
+	}
+
+	return i.validateAndSetEndpoint(domain)
+}
+
+func (i *apiCmdInputs) validateAndSetMethod() error {
+	for _, validMethod := range apiValidMethods {
+		if i.RawMethod == validMethod {
+			i.Method = i.RawMethod
+			return nil
+		}
+	}
+
+	return fmt.Errorf(
+		"invalid method given: %s, accepting only %s",
+		i.RawMethod,
+		strings.Join(apiValidMethods, ", "),
+	)
+}
+
+func (i *apiCmdInputs) validateAndSetData() error {
+	var data []byte
+
+	if i.RawData != "" {
+		data = []byte(i.RawData)
+	}
+
+	pipedRawData := iostream.PipedInput()
+	if len(pipedRawData) > 0 && data == nil {
+		data = pipedRawData
+	}
+
+	if len(pipedRawData) > 0 && len(i.RawData) > 0 {
+		i.renderer.Warnf(
+			"JSON data was passed using both the flag and as piped input. " +
+				"The command will use only the data from the flag.",
```

Comment:
```suggestion
				"The Auth0 CLI will use only the data from the flag.",
```
From our perspective it's the command's implementation, from the user's perspective it's the Auth0 CLI.

Reply:
[`f401dfd` (#531)](https://github.com/auth0/auth0-cli/pull/531/commits/f401dfd09705e3d6960f19f7f9a6333cbcdadba4)